src/rtutils.c

```diff
@@ -777,6 +804,42 @@ static int jl_static_is_function_(jl_datatype_t *vt) JL_NOTSAFEPOINT {
     return 0;
 }
 
+static size_t jl_static_show_float(JL_STREAM *out, double v,
+                                   jl_datatype_t *vt) JL_NOTSAFEPOINT
+{
+    size_t n = 0;
+    // TODO: non-canonical NaNs do not round-trip
+    // TOOD: BFloat16
+    const char *size_suffix = vt == jl_float16_type ? "16" :
+                              vt == jl_float32_type ? "32" :
+                                                      "";
+    // Digits required to print p significand bits: ceil(p * log(10, 2))
+    //   Float16   4
+    //   Float32   8
```

Comment:
These seem unlikely correct, since the standard given results are one higher for these:
https://github.com/JuliaLang/julia/pull/57977/files#r2028868985
```suggestion
    //   Float16   5
    //   Float32   9
```

Reply:
Corrected formula from https://dl.acm.org/doi/pdf/10.1145/93542.93559: 

$N = 2 + \left\lfloor\frac{n}{\log_b B}\right\rfloor$

Reply:
```
function find_counterexample(T, prec, r)
    buf = zeros(UInt8, 32)
    for i=r
        x = reinterpret(T, i)
        isnan(x) && continue
        n = @ccall snprintf(buf::Ptr{UInt8}, 32::Csize_t, "%.*g"::Cstring; prec::Cint, x::Cdouble)::Csize_t
        ok, y = @ccall jl_try_substrtof(buf::Ptr{UInt8}, 0::Csize_t, n::Csize_t)::Tuple{Bool,Float32}
        ok && x === T(y) || return x
    end
end

@assert find_counterexample(Float16, 4, 0x0000:0xffff) === Float16(0.00010014)
@assert find_counterexample(Float16, 5, 0x0000:0xffff) === nothing

@assert find_counterexample(Float32, 8, 0x00000000:0xffffffff) === 1.00000075f-36
@assert find_counterexample(Float32, 9, 0x00000000:0xffffffff) === nothing
```